app/src/test/java/me/jorgecastillo/hiroaki/MockForeverTests.kt

```diff
@@ -0,0 +1,122 @@
+    fun shouldRespondMultipleTimes() {
+        server.whenever(Method.GET, "v2/top-headlines")
+            .thenRespondForever(success(jsonBody = fileBody("GetNews.json")))
+
+        repeat(3) {
+            runBlocking { dataSource.getNews() }
+        }
+
+        server.verify("v2/top-headlines").called(times(count = 3))
+    }
+
+    @Test
+    fun shouldDispatchMultipleTimes() {
+        server.whenever(Method.GET, "v2/top-headlines")
+            .thenDispatchForever { request ->
+                success(
+                    jsonBody = inlineBody(
+                        """{
+                              "status": "ok",
+                              "totalResults": 2342,
+                              "articles": [
+                                {
+                                  "source": {
+                                    "id": ${request.path.length},
+                                    "name": "Lifehacker.com"
+                                  },
+                                  "author": "Jacob Kleinman",
+                                  "title": "How to Get Android P's Screenshot Editing Tool on Any Android Phone",
+                                  "description": "Last year, Apple brought advanced screenshot editing tools to the iPhone with iOS 11, and, this week, Google fired back with a similar Android feature called Markup. The only catch is that this new tool is limited to Android P, which launches later this year …",
+                                  "url": "https://lifehacker.com/how-to-get-android-ps-screenshot-editing-tool-on-any-an-1823646122",
+                                  "urlToImage": "https://i.kinja-img.com/gawker-media/image/upload/s--Y-5X_NcT--/c_fill,fl_progressive,g_center,h_450,q_80,w_800/nxmwbkwzoc1z1tmak7s4.jpg",
+                                  "publishedAt": "2018-03-09T20:30:00Z"
+                                }
+                              ]
+                            }
+                        """
+                    )
+                )
+            }
+
+        repeat(3) {
+            runBlocking { dataSource.getNews() }
+        }
+
+        server.verify("v2/top-headlines").called(times(count = 3))
+    }
+
+    @Test(expected = IOException::class)
+    fun regularResponseShouldOverrideRespondForever() {
+        server.whenever(Method.GET, "v2/top-headlines")
+            .thenRespond(error())
+            .thenRespondForever(success(jsonBody = fileBody("GetNews.json")))
+
+        runBlocking {
+            dataSource.getNews()
+        }
+
+        server.verify("v2/top-headlines").called()
+    }
+
+    @Test
+    fun secondResponseForeverShouldOverrideTheFirst() {
+        server.whenever(Method.GET, "v2/top-headlines")
+            .thenRespondForever(error())
+            .thenRespondForever(success(jsonBody = fileBody("GetNews.json")))
+
+        runBlocking {
+            dataSource.getNews()
+        }
+
+        server.verify("v2/top-headlines").called()
```

Comment:
Same here, we should verify the second one is the one responded for N calls and the first one is ignored